azurelinuxagent/ga/monitor.py

```diff
@@ -177,8 +177,8 @@ def init_sysinfo(self):
         try:
             ram = self.osutil.get_total_mem()
             processors = self.osutil.get_processor_cores()
-            self.sysinfo.append(TelemetryEventParam("RAM", ram))
-            self.sysinfo.append(TelemetryEventParam("Processors", processors))
+            self.sysinfo.append(TelemetryEventParam("RAM", int(ram)))
```

Comment:
Shouldn't we be respecting the schema in Kusto?


```
cluster("Rdos").database("rdos").GuestAgentExtensionEvents
| take 1
| project RAM, Processors, ImageOrigin | getschema 


ColumnName   | ColumnOrdinal  | DataType       | ColumnType
RAM          | 0              | System.Int64   | long
Processors   | 1              | System.Int64   | long
ImageOrigin  | 2              | System.String  | string


```

Reply:
We translate the types in [wire.py](https://github.com/Azure/WALinuxAgent/blob/develop/azurelinuxagent/common/protocol/wire.py#L507-L523) while generating the event.

```
def event_param_to_v1(param):
    param_format = '<Param Name="{0}" Value={1} T="{2}" />'
    param_type = type(param.value)
    attr_type = ""
    if param_type is int:
        attr_type = 'mt:uint64'
    elif param_type is str:
        attr_type = 'mt:wstr'
    elif ustr(param_type).count("'unicode'") > 0:
        attr_type = 'mt:wstr'
    elif param_type is bool:
        attr_type = 'mt:bool'
    elif param_type is float:
        attr_type = 'mt:float64'
    return param_format.format(param.name,
                               saxutils.quoteattr(ustr(param.value)),
                               attr_type)
```

[In the example XML generated](https://github.com/Azure/WALinuxAgent/blob/develop/tests/ga/test_monitor.py#L433-L434) - :

```
'<Param Name="RAM" Value="10000" T="mt:uint64" />' \
'<Param Name="Processors" Value="4" T="mt:uint64" />' \
```